src/kOS/Suffixed/CraftTemplate.cs

```diff
@@ -0,0 +1,141 @@
+                {
+                    case ("vab"):
+                        EditorDriver.editorFacility = EditorFacility.VAB;
+                        break;
+
+                    case ("sph"):
+                        EditorDriver.editorFacility = EditorFacility.SPH;
+                        break;
+
+                    default:
+                        throw new KOSException("Failed to load craft.  Facility " + facility + " not recognized, expected \"VAB\" or \"SPH\"");
+                }
+                var path = ShipConstruction.GetSavePath(name);
+                var template = ShipConstruction.LoadTemplate(path);
+                if (template == null)
+                    throw new KOSException("Failed to load craft named \"" + name + "\" from given path:\n  " + path);
+                return new CraftTemplate(path, template);
+            }
+            throw new KOSException("Failed to load craft named " + name + " from facility " + facility);
+        }
+
+        public static ListValue GetAllTemplates()
+        {
+            ListValue ret = new ListValue();
+            var files = GetAllPathStrings();
+            foreach (string path in files)
+            {
+                var template = ShipConstruction.LoadTemplate(path);
+                if (template != null)
+                {
+                    ret.Add(new CraftTemplate(path, template));
+                }
+            }
+            return ret;
+        }
+
+        private static List<string> GetAllPathStrings()
+        {
+            var ret = new List<string>();
+            string path1 = KSPUtil.GetOrCreatePath("Ships/VAB");
+            string path2 = KSPUtil.GetOrCreatePath("Ships/SPH");
+            string path3 = KSPUtil.GetOrCreatePath("saves/" + HighLogic.SaveFolder + "/Ships/VAB");
+            string path4 = KSPUtil.GetOrCreatePath("saves/" + HighLogic.SaveFolder + "/Ships/SPH");
+            if (HighLogic.CurrentGame.Parameters.Difficulty.AllowStockVessels)
+            {
+                ret.AddRange(Directory.GetFiles(path1));
+                ret.AddRange(Directory.GetFiles(path2));
+            }
+            ret.AddRange(Directory.GetFiles(path3));
+            ret.AddRange(Directory.GetFiles(path4));
+            return ret;
+        }
+
+        private string path;
+        private ShipTemplate template;
+
+        public CraftTemplate(string filePath)
+        {
+            path = filePath;
+            template = ShipConstruction.LoadTemplate(filePath);
+            if (template == null)
+                throw new KOSException("Failed to load template from given path:\n  " + filePath);
+            InitializeSuffixes();
+        }
+
+        public CraftTemplate(string filePath, ShipTemplate temp)
+        {
+            path = filePath;
+            template = temp;
+            if (template == null)
+                throw new KOSException("Failed to load template from given path:\n  " + filePath);
+            InitializeSuffixes();
```

Comment:
I've updated it (though these comments aren't hidden because the associated line itself wasn't modified).